packages/core/types/client.d.ts

```diff
@@ -1,34 +1,58 @@
-import Breadcrumb from "./breadcrumb";
 import * as common from "./common";
-import Report from "./report";
+import Breadcrumb from "./breadcrumb";
+import Event from "./event";
 import Session from "./session";
 
 declare class Client {
-  public app: object;
-  public device: object;
-  public context: string | void;
-  public config: common.IConfig;
-  public user: object;
-  public metaData: object;
-
-  public BugsnagReport: typeof Report;
-  public BugsnagBreadcrumb: typeof Breadcrumb;
-  public BugsnagSession: typeof Session;
-
-  public use(plugin: common.IPlugin, ...args: any[]): Client;
-  public getPlugin(name: string): any;
-  public setOptions(opts: common.IConfig): Client;
-  public configure(schema?: common.IConfigSchema): Client;
-  public delivery(delivery: common.IDelivery): Client;
-  public logger(logger: common.ILogger): Client;
-  public sessionDelegate(sessionDelegate: common.ISessionDelegate): Client;
+  private constructor();
+
+  // reporting errors
   public notify(
```

Comment:
We try to avoid overloads where possible. Since they are not a language feature of javascript,  it makes the implementation more difficult as you have to do a bunch of duck-typing and resulting logic on the arguments that were received.

Instead we support the `error` parameter being an object which has the following properties `{ name, message }`, so I guess that's a deviation and should be noted as such.